python/ray/util/client/worker.py

```diff
@@ -110,8 +112,9 @@ def __init__(self,
                 grpc.channel_ready_future(self.channel).result(timeout=timeout)
                 # The HTTP2 channel is ready. Wrap the channel with the
                 # RayletDriverStub, allowing for unary requests.
-                self.server = ray_client_pb2_grpc.RayletDriverStub(
-                    self.channel)
+                server = ray_client_pb2_grpc.RayletDriverStub(self.channel)
+                # Require metadata with all requests
+                self.server = require_metadata(server)
```

Comment:
Found it easiest to just do this at runtime.
Added a test for `require_metadata`.

Reply:
Can't we just make this a testing where we go over all the functions and verify that we can pass metadata to them? 
I find the current code a little bit hard to read and can leave hidden bugs.

Reply:
The issue was that it was possible to not pass metadata.
I think we can just remove this logic and we'll be fine.

Can you explain more precisely what you had in mind for testing?

Reply:
This is indeed hard to read and prone to bugs.
Reverting.